### PR TITLE
Add Image support

### PR DESCRIPTION
**Pull request by novakov-alexey**

### CODE REVIEW

**Comment by novakov-alexey** [comment]
@alexarchambault Documentation tests seem not relevant for Image Trace due to outdated documentation fork. plotly-scala would require to update documentation to the latest Plotly.js state. However, Image trace is working fine:

```scala
import plotly.element.HoverInfo
import plotly.element.HoverInfo._

val z = Seq(
    Seq(Seq(0, 33, 50), Seq(0, 66, 50), Seq(0, 100, 50)),
    Seq(Seq(90, 33, 50), Seq(90, 66, 50), Seq(90, 100, 50)),
    Seq(Seq(180, 33, 50), Seq(180, 66, 50), Seq(180, 100, 50)),
    Seq(Seq(270, 33, 50), Seq(270, 66, 50), Seq(270, 100, 50)))

Image(
    z = z,
    x0=Some(0.05),
    y0=Some(0.05),
    hoverinfo = Some(HoverInfo(X, Y, Z)),
    colormodel = Some("rgb")
).plot()
```

**Comment by alexarchambault** [comment]
@novakov-alexey I'm about to update the documentation sub-module, I should open a PR for it, either later today or in the coming days. That should allow to add tests for this.

Just ping me if I forget to come back here once it's done.

**Comment by novakov-alexey** [comment]
Great, looking forward to that!

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 target/
 .bsp/
+.idea
```
Comment (alexarchambault) [inline .gitignore:3]: Can this be added in a separate commit, to make the git history cleaner?

File: README.md [hunk 2]
```diff
@@ -28,10 +28,11 @@ It is published for both scala 2.12 and 2.13.
 
 ### From almond
 
-Add the `org.plotly-scala::plotly-almond:0.8.0` dependency to the notebook. (Latest version: [![Maven Central](https://img.shields.io/maven-central/v/org.plotly-scala/plotly-render_2.13.svg)](https://maven-badges.herokuapp.com/maven-central/org.plotly-scala/plotly-render_2.13))
+Add the `org.plotly-scala::plotly-almond:0.8.1` dependency to the notebook. (Latest version: [![Maven Central](https://img.shields.io/maven-central/v/org.plotly-scala/plotly-render_2.13.svg)](https://maven-badges.herokuapp.com/maven-central/org.plotly-scala/plotly-render_2.13))
 Then initialize plotly-scala, and use it, like
+
 ```scala
-import $ivy.`org.plotly-scala::plotly-almond:0.8.0`
+import $ivy.`org.plotly-scala::plotly-almond:0.8.1`
```
Comment (alexarchambault) [inline README.md:35]: Same as above, let's edit the README in a separate commit too.

**Review by alexarchambault — COMMENTED** [review]
I only left some minor comments.

I also updated the documentation sub-module in `master`, feel free to rebase, and enable relevant tests for images.

**Comment by novakov-alexey** [comment]
There is a small problem that I do not know how to resolve on JSON encoding level.

If we want to support 3-channel and 4 channel color models like RGB and RGBA, we might need to model colors in `z` property as some case classes. For example:

```scala
abstract class ZColor with ....
Color3(c1: Int, c2: Int, c3: Int) extends ZColor
// note that last property is Double, all other are Ints
Color4(c1: Int, c2: Int, c3: Int, alpha: Double) extends ZColor 
```
I tried to implement it, but failed to render these case classes without `class name` in the resulting JSON string. Currently, `Trace` encoder renders every type by printing all nested property type names/keys. That breaks [z property](https://plotly.com/javascript/reference/image/#image-z). Expected JavaScript format of `z` is z: [[ [123,123,123], [.....] ]]

Latets API I came up with:

```scala
import plotly.element.HoverInfo._
import plotly.element.ColorModel._
import plotly.element.Element._

val colors = Seq(
    Seq(Seq(0d, 33d, 50d), Seq(0d, 66, 50), Seq(0d, 100, 50)),
    Seq(Seq(90d, 33d, 50d), Seq(90d, 66, 50), Seq(90d, 100, 50)),
    Seq(Seq(180d, 33d, 50d), Seq(180d, 66, 50), Seq(180d, 100, 50)),
    Seq(Seq(270d, 33d, 50d), Seq(270d, 66, 50), Seq(270d, 100, 50)))

Image(z = colors)
  .withX0(0.05)
  .withY0(0.05)
  .withHoverinfo(HoverInfo(X, Y, Z, HoverInfo.Color))
  .withColormodel(RGB)
  .plot()
```
It works, but requires Double numbers for RGB, which is weird.

**Comment by novakov-alexey** [comment]
New PR is here: #274
